CMakeLists.txt

```diff
@@ -98,14 +98,12 @@ set(scylla_sources
     auth/service.cc
     auth/standard_role_manager.cc
     auth/transitional.cc
-    bytes.cc
```

Comment:
Why are these files added to the utils lib?
If you think they belong there conceptionally, please move them to the `utils/` directory first (in a separate PR please).

Reply:
i was trying to fix the dependencies of the tests. as in the C++20 module era, we are likely have a bunch of modules, and tests are going to import (some of) them for exercising the exposed functionalities. and the tests are not likely to need to link against the whole scylla blob in a whole sale for testing a tiny piece of functionality.

but i failed, this dependencies between the translation units in `scylla_sources` and the ones in the subdirectories is not a tree. sometimes, they form circular graph.

now that the change does not work at all. i will drop this change. and keep it in `scylla_sources` in this pull request, and revisit this problem later.

Reply:
> sometimes, they form circular graph

Yes, this is a known problem that needs to be sorted out before we use modules, which requires a DAG dependencies.